dispatch/static/manager/src/js/actions/ArticlesActions.js

```diff
@@ -1,210 +1,59 @@
-  }
-
-  data.section_id = data.section
-
-  data.author_ids = data.authors
-
-  data.tag_ids = data.tags
-
-  data.topic_id = data.topic
-
-  data.template_id = data.template
-  delete data.template
-
-  if (data.featured_image) {
-    data.featured_image.image_id = data.featured_image.image
-  }
-
-  return data
-}
-
-export function fetchArticles(token, query) {
-  return {
-    type: types.FETCH_ARTICLES,
-    payload: DispatchAPI.articles.fetchArticles(token, query)
-      .then(function(json) {
-        return {
-          count: json.count,
-          results: normalize(json.results, arrayOf(articleSchema))
-        }
-      })
-  }
-}
-
-export function fetchArticle(token, articleId, params) {
-  return {
-    type: types.FETCH_ARTICLE,
-    payload: DispatchAPI.articles.fetchArticle(token, articleId, params)
-      .then( json => normalize(json, articleSchema) )
-  }
-}
-
-export function setArticle(article) {
-  return {
-    type: types.SET_ARTICLE,
-    payload: normalize(article, articleSchema)
-  }
-}
-
-export function saveArticle(token, articleId, data) {
-
-  data = preparePayload(data)
-
-  return {
-    type: types.SAVE_ARTICLE,
-    payload: DispatchAPI.articles.saveArticle(token, articleId, data)
-      .then( json => normalize(json, articleSchema) )
-  }
-}
-
-export function createArticle(token, data) {
-
-  data = preparePayload(data)
+  prepareData(data) {
+    if (data._content) {
+      // Convert article contentState to JSON array
+      data.content = ContentStateHelper.toJSON(data._content)
+      delete data._content
+    }
 
-  return {
-    type: types.CREATE_ARTICLE,
-    payload: DispatchAPI.articles.createArticle(token, data)
-  }
-}
+    data.section_id = data.section
 
-export function publishArticle(token, articleId, data) {
-
-  data = preparePayload(data)
-
-  return function(dispatch) {
-
-    dispatch({ type: `${types.PUBLISH_ARTICLE}_PENDING` })
-
-    DispatchAPI.articles.saveArticle(token, articleId, data)
-      .then(() => DispatchAPI.articles.publishArticle(token, articleId))
-      .then(json => normalize(json, articleSchema))
-      .then(payload => {
-        dispatch({
-          type: `${types.PUBLISH_ARTICLE}_FULFILLED`,
-          payload: payload
-        })
-      })
-      .catch(error => {
-        dispatch({
-          type: `${types.PUBLISH_ARTICLE}_REJECTED`,
-          payload: error
-        })
-      })
-  }
-}
-
-export function unpublishArticle(token, articleId) {
-  return {
-    type: types.UNPUBLISH_ARTICLE,
-    payload: DispatchAPI.articles.unpublishArticle(token, articleId)
-      .then( json => normalize(json, articleSchema) )
-  }
-}
+    data.author_ids = data.authors
 
-export function previewArticle(token, articleId, data) {
+    data.tag_ids = data.tags
 
-  data = preparePayload(data)
+    data.topic_id = data.topic
 
-  return function(dispatch) {
-    dispatch({ type: `${types.SAVE_ARTICLE}_PENDING` })
+    data.template_id = data.template
```

Comment:
off topic of PR, and i haven't contributed to this development 🙃 , so sorry for being obnoxious, but what is up with the funky key-reassignment in this `prepareData` function? could the python (de)serializers, say, be changed to do this name translation?